mixed threshold docs

woodsl92 opened the pull request:
Added description of mixed-species cell-caller behaviour, and L-plot examples.

ImogenCarruthersCS commented on README.md:
```diff
@@ -582,12 +583,27 @@ It does this by assessing the distribution of total counts per barcode across al
+### Mixed-species cell calling
+When working with a mixed-species sample Cell Caller will identify 2 candidate subpopulations of barcodes (1 for each species), based on the proportion of reads for that barcode which map to each species. E.g. in a sample containing a mixture of human and mouse cells, Cell Caller identifies candidate human-associated barcodes as those where the number of human-mapped reads is greater than the num ...
+
+The Cell Caller algorithm is run on each of the candidate subpopulations to set a minimum count threshold for each species. Once the species-specific minimum count thresholds are defined we use them to split the entire pool of cells into 4 groups, as depicted in the L-plot below. 
```
Please double check that the cell caller section uses exact and correct language throughout. For example, here, this should refer to splitting the entire pool of barcodes into 4 groups, not cells.

ImogenCarruthersCS commented on README.md:
```diff
@@ -582,12 +583,27 @@ It does this by assessing the distribution of total counts per barcode across al
+
+<img src="docs/images/mixed_species_appropriate_threshold_L_plot.png" alt="Mixed-species L-plot with appropriate thresholds" width="600" height="400"/>
+
+In some instances the mouse or human (or both) minimum count threshold may be set inappropriately. In the example below the human threshold has been set too low, resulting in cells that have predominantly human counts being wrongly defined as multiplets. In this case we would suggest to [set the human Cell Caller threshold manually.](#setting-manual-cell-caller-thresholds)
```
This should read: 'resulting in cells that have predominantly mouse counts ...', not human.

ImogenCarruthersCS replied:
Also, suggest to rephrase the first sentence to improve readability:
"In some instances, the mouse or human threshold (or both) may be set inappropriately…”

woodsl92 commented:
Made those changes suggested and did a final read-through.
If you're happy let me know, I'll squash those commits

ImogenCarruthersCS commented on README.md:
```diff
@@ -578,16 +579,33 @@ Contains the bam files output from STAR and associated mapping log files.
+- candidate mouse cells, where mouse counts &gt; human counts
+- candidate human cells, where human counts &gt; mouse counts
+
+The Cell Caller algorithm is run on each of the candidate subpopulations to set a minimum count threshold for each species. Once the species-specific minimum count thresholds are defined we use them to split the entire pool of barcodes into 4 groups, as depicted in the L-plot below. 
```
This is still missing the info about what distributions the cell caller logic is run on (mouse counts for candidate mouse cells, and human counts for candidate human cells). I mentioned previously that this is described nicely in the python script. Please go back to this and transfer the relevant information over.

woodsl92 replied:
Apologies for the misunderstanding, I've rewritten the 'theory of cell caller' and 'mixed species' sections to more closely follow the wording the docustring, and to be more clear and concise. Also checked that the links and images are as expected.

woodsl92 commented:
Apologies for the misunderstanding, I've rewritten the 'theory of cell caller' and 'mixed species' sections to more closely follow the wording the docustring, and to be more clear and concise. Also checked that the links and images are as expected.

Squashed the commits from this PR, editing the cell caller docs into 1.

Hope this is closer to the mark now, but happy to make any further changes if necessary.

ImogenCarruthersCS commented on README.md:
```diff
@@ -577,26 +578,46 @@ Contains the bam files output from STAR and associated mapping log files.
+<br><br>
+SimpleCell&trade; libraries typically produce a multimodal distribution, as shown in the example below. The right-most peak in this distribution (D) represents cell-associated barcodes, while the peaks on the left (A, B) contain barcodes with fewer reads, representing noise-associated barcodes.<br><br>
+<img src="docs/images/typical_cell_caller.png" alt="Typical Cell Caller plot" width="600" height="400"/>
+<br><br>The count threshold is determined by finding the first local minima of the probability density (i.e. the lowest point, C) between the cell and noise peaks. The count threshold is then used to call cells. Any cellular barcode with counts below the threshold is called as noise, and any barcode with counts above the threshold is called as a cell.
```
One minor rephrase just for total clarity (and I apologise for not picking this one up earlier - I've just read it start to finish with fresh eyes and this is the only thing that stood out in terms of flow) -

Please could this be changed to "The count threshold is determined by finding the first local minima of the probability density (i.e. the lowest point, C) above the minimum_count_threshold (set to 100 counts by default). The count threshold is then ..."

ImogenCarruthersCS reviewed (changes requested):
Hi Laura. Thank you for making those changes. This looks great now. I only have one minor rephrase (and I apologise for not requesting this earlier). Once that is done, I think this is good to ship.
